cleanlab/outlier.py

```diff
@@ -0,0 +1,534 @@
+
+        """
+        scores = self._shared_fit(
+            features=features,
+            pred_probs=pred_probs,
+            labels=labels,
+            verbose=verbose,
+        )
+
+        if scores is None:  # Fit was called on already fitted object so we just score vals instead
+            scores = self.score(features=features, pred_probs=pred_probs)
+
+        return scores
+
+    def fit(
+        self,
+        *,
+        features: Optional[np.ndarray] = None,
+        pred_probs: Optional[np.ndarray] = None,
+        labels: Optional[np.ndarray] = None,
+        verbose: bool = True,
+    ):
+        """
+        Fits an estimator following optional parameters specified during object initialization. One of `features` or `pred_probs`
+        needs to be passed in to fit estimator object.
+
+        If `features` are passed in a `NearestNeighbors` object is fit. For details on the object see
+        :py:class:`OutOfDistribution <cleanlab.outlier.OutOfDistribution>`.
+
+        If `pred_probs` and 'labels' are passed in a `confident_thresholds` np.ndarray is fit. For details on
+        'confindent_thresholds` see :py:class:`OutOfDistribution <cleanlab.outlier.OutOfDistribution>`.
+
+        Parameters
+        ----------
+        features : np.ndarray, optional
+          Feature array of shape ``(N, M)``, where N is the number of examples and M is the number of features used to represent each example.
+          All features should be numeric. For unstructured data (eg. images, text, categorical values, ...), you should provide
+          vector embeddings to represent each example (e.g. extracted from some pretrained neural network).
+
+        pred_probs : np.ndarray, optional
+           An array of shape ``(N, K)`` of model-predicted probabilities,
+          ``P(label=k|x)``. Each row of this matrix corresponds
+          to an example `x` and contains the model-predicted probabilities that
+          `x` belongs to each possible class, for each of the K classes. The
+          columns must be ordered such that these probabilities correspond to
+          class 0, 1, ..., K-1.
+
+          **Caution**: `pred_probs` from your model must be out-of-sample!
+          You should never provide predictions on the same examples used to train the model,
+          as these will be overfit and unsuitable for finding label-errors.
+          To obtain out-of-sample predicted probabilities for every datapoint in your dataset, you can use :ref:`cross-validation <pred_probs_cross_val>`.
+          Alternatively it is ok if your model was trained on a separate dataset and you are only evaluating
+          data that was previously held-out.
+
+        labels : np.ndarray, optional
+          A discrete vector of noisy labels, i.e. some labels may be erroneous of shape ``(N,)``.
```

Comment:
"noisy labels, i.e. some labels may be erroneous" -> "given labels for the data"